### ISSUE TITLE
netstream.transmit: Failed to write string to stream, maximum stream size achived

### ISSUE THREAD

**Issue by X-Coder** [Rusketh/ExpAdv2#313]
Error:
Expression Advanced 2 - Uncatched Exception (serverside):
stream - Failed to write string to stream, maximum stream size achived (128)

I try to send multiple strings (on each button press) from one "controller" ea2 to multiple other ea2-screen clients and get this error when I have pressed the send-text button like 30 times. Currently I have spawned only one screen + the controller one.

I do need transmit(...) in this case, right? broadcast is working only to send stuff between the same chip.

How do I get around this error? 
(Stream = stream() // trying to start a new stream does not work too, I can't write to it then.)

code of controller ea2:
```LUA
server{
    print("Server")
    array screens = findByClass("expadv_screen")    
    input number Button      
    netstream Stream = stream()
    event trigger(string PortName, string Type){
        if(PortName == "Button" && Button==1){
            foreach (number key; entity screen: screens) {
                if(screen.owner() == owner()){                    
                    Stream.writeString("Testmessage")
                    Stream.transmit(screen, "start")                    
                }
            }            
        }
    }        
}
```

code of screens ea2:
```LUA
server{
    print("server-screen")
    netstream clientStream = stream()
    hookStream("start", function(stream stream){
        print("server: start hook recieved:" + stream.readString())
        clientStream.writeString(stream.readString())
        clientStream.netBroadcast("setMessage")
    })    
}
client{
    string message
    netReceive("setMessage", function(netstream clientStream){
        message += "\n" + clientStream.readString()
        print("client: start hook recieved:" + message)
        
    })
    
    //this hook seems not get called
    /*hookStream("start", function(stream clientStream){
        message += "\n" + clientStream.readString()
        print("client: start hook recieved:" + message)        
    })*/
    
    int linePosition
    event drawScreen(number a, number b){
        if(message){
            print("newMessage drawn")
            linePosition+=30
            noFrameRefresh(false)
            if(linePosition>=30*15){
                clearScreen()
                linePosition=0
            }            
            drawText(vec2(30,linePosition), message)    
            noFrameRefresh(true)
            message = ""
        }
    }
}
```

**Comment by bigdogmat** [comment]
This is happening because streams have a limit of 128 entries, and sending a stream doesn't reset that amount. The fix would be to recreate the stream each time you send it, which you should be doing anyway

**Comment by X-Coder** [comment]
I don't want to send 128 entries, only one. I tried netstream Stream = stream() before inside my for loop to recreate the whole stream, but I forgot to do it on the second chip on my clientStream var too. This seems to work now. The error message was because of my second one and I didn't have seen that its this one causing it.

Only odd thing was in hookStream I had to write netstream Stream = stream without brackets to get it working.
